cvmfs/swissknife_pull.cc

```diff
@@ -76,25 +76,21 @@ string              *preload_cachedir = NULL;
 }  // anonymous namespace
 
 
-static bool Peek(const string &remote_path, const char suffix)
-{
+static bool Peek(const string &remote_path) {
   if (preload_cache) {
     // Strip "data"
+    // TODO(rene): Fix that! It's ugly and error prone!
     return FileExists(*preload_cachedir + remote_path.substr(4));
```

Comment:
I haven't tried it but I think things will break here.  When the cache is preloaded, path names are always without suffix.  This is different from Stratum 1 replication.